### ISSUE TITLE
A SearchCollection index error occurs when a search is performed before the search index is built.

### ISSUE THREAD

**Issue by kieckhafer** [reactioncommerce/reaction#1396]
### Expected behavior

When the app is initialized, no errors should occur if a search is performed before ProductSearch collection is built.
### Actual Behavior

If a search is performed in the time between the app initialization and ProductSearch collection being built, an error is thrown:

```
I20160916-10:52:40.508(-7)? Exception from sub SearchResults id Qm4bpXZ7Zc3cDdC7P Error: Exception while polling query {"collectionName":"ProductSearch","selector":{"shopId":"J8Bhq3uTtdgwZx3rz","$text":{"$search":"te"},"isVisible":true},"options":{"transform":null,"fields":{"score":{"$meta":"textScore"},"title":1,"hashtags":1,"description":1,"handle":1,"price":1},"sort":{"score":{"$meta":"textScore"}},"limit":99}}: text index required for $text query
I20160916-10:52:40.508(-7)?     at Object.Future.wait (/Users/aaronjudd/.meteor/packages/meteor-tool/.1.4.1_1.ivuj7v++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/server-lib/node_modules/fibers/future.js:449:15)
I20160916-10:52:40.508(-7)?     at ObserveMultiplexer.addHandleAndSendInitialAdds (packages/mongo/observe_multiplex.js:55:23)
I20160916-10:52:40.509(-7)?     at MongoConnection._observeChanges (packages/mongo/mongo_driver.js:1271:15)
I20160916-10:52:40.509(-7)?     at Cursor.observeChanges (packages/mongo/mongo_driver.js:946:22)
I20160916-10:52:40.509(-7)?     at Function.Mongo.Collection._publishCursor (packages/mongo/collection.js:344:30)
I20160916-10:52:40.509(-7)?     at Cursor._publishCursor (packages/mongo/mongo_driver.js:927:27)
I20160916-10:52:40.509(-7)?     at Subscription._publishHandlerResult (packages/ddp-server/livedata_server.js:1068:13)
I20160916-10:52:40.510(-7)?     at Subscription._runHandler (packages/ddp-server/livedata_server.js:1041:10)
I20160916-10:52:40.510(-7)?     at Session._startSubscription (packages/ddp-server/livedata_server.js:845:9)
I20160916-10:52:40.510(-7)?     at Session.sub (packages/ddp-server/livedata_server.js:617:12)
I20160916-10:52:40.510(-7)?     at packages/ddp-server/livedata_server.js:551:43
I20160916-10:52:40.510(-7)?     - - - - -
I20160916-10:52:40.511(-7)?     at PollingObserveDriver._pollMongo (packages/mongo/polling_observe_driver.js:161:11)
I20160916-10:52:40.511(-7)?     at Object.task (packages/mongo/polling_observe_driver.js:90:12)
I20160916-10:52:40.511(-7)?     at [object Object]._.extend._run (packages/meteor/fiber_helpers.js:147:1)
I20160916-10:52:40.511(-7)?     at packages/meteor/fiber_helpers.js:125:1
```
### Steps to Reproduce the Behavior

`rrr`
IMMEDIATELY perform a search. The error only occurs in a 10-15ish second span between initialization and the collection being built.

```
I20160916-11:22:59.864(-7)? 18:22:59.856Z  INFO Reaction: No ProductSearch records found. Adding build ProductSearch Collection to jobs
I20160916-11:22:59.930(-7)? 18:22:59.906Z  INFO Reaction: Reaction.init() has run
I20160916-11:22:59.931(-7)? 18:22:59.906Z  INFO Reaction: Reaction initialization finished.
=> Started your app.

=> App running at: http://localhost:3000/
I20160916-11:23:13.389(-7)? 18:23:13.385Z  INFO Reaction: ProductSearch collection (re)built successfully.
```

**Comment by kieckhafer** [comment]
👍 No longer seeing the error when searching before search index isn't yet built.

**Comment by zenweasel** [comment]
Closed via #1621
